juju/juju
# [JUJU-2259] Add errors to lease store interface function returns

The current lease store interface suits the Raft-backed store (FSM). The Raft FSM is able to return lease listings and pinned leases directly from memory without the possibility of errors, so the interface signatures for such functionality do not include error returns.

This is not compatible with a Dqlite-backed lease store, which _can_ encounter errors for these operations.

Here we add errors where they are missing from the interface method returns, and follow this though all implementations and downstream consumers.

Changes to the RPC payload for querying pinned leases is acceptable here, because the API call is not yet recruited anywhere.

As a drive by, the dead lease manager functionality and tests are removed - such functionality is vestigial and not used.

## QA steps

Just check that lease functionality still works - deploy some workloads and check that we get leaders:
- `make go-install`.
- Bootstrap to LXD.
- `juju enable-ha`.
- `juju add-model default && deploy ubuntu -n 3`.
- Check that everything settles without issue.

## Documentation changes

None.

## Bug reference

N/A
